### ISSUE TITLE
pex can resolve incompatible wheels with the 'u'/UCS4 abi flag

### ISSUE THREAD

**Issue by kwlzn** [pantsbuild/pex#303]
if presented with a `find-links` repo with deps with a 'u' flag set in the abi field (indicating UCS4/wide unicode), pex can erroneously resolve these wheels in a UCS2 interpreter producing a pex that will break at runtime with unicode errors.

given there's a general, reasonable expectation of buildtime interpreter compatibility vs runtime interpreter compatibility the pex resolver should check for unicode compatibility during resolves and limit its scope to only compatible wheels at the abi level.

repro:

```
[illuminati unicode_wheels]$ ls -1 test
cffi-1.7.0-cp27-cp27mu-macosx_10_10_x86_64.whl
pycparser-2.14-py2-none-any.whl
[illuminati unicode_wheels]$ pex
Python 2.7.10 (default, Dec 16 2015, 14:09:45) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> import sysconfig; sysconfig.get_config_var('Py_UNICODE_SIZE')
2
>>> 
[illuminati unicode_wheels]$ pex --disable-cache --no-build --no-pypi -f ./test -vvvvvvvvv cffi==1.7.0 -o /tmp/throwaway.pex
pex: crawling link i=0 link=Link('file:///private/tmp/unicode_wheels/test') follow_links=False                         
pex: Building pex :: Resolving distributions :: Fetching file:///private/tmp/unicode_wheels/test/cffi-1.7.0-cp27-cp27mu-macosx_10_10_x86pex: Building pex :: Resolving distributions :: Fetching file:///private/tmp/unicode_wheels/test/cffi-1.7.0-cp27-cp27mu-macosx_10_10_x86pex: Building pex :: Resolving distributions :: Translating /private/var/folders/3t/xkwqrkld4xxgklk2s4n41jb80000gn/T/tmpZVta6g/cffi-1.7.pex: Building pex :: Resolving distributions :: Fetching file:///private/tmp/unicode_wheels/test/pycparser-2.14-py2-none-any.whl        pex: Building pex :: Resolving distributions :: Fetching file:///private/tmp/unicode_wheels/test/pycparser-2.14-py2-none-any.whl :: Fetcpex: Building pex :: Resolving distributions :: Translating /private/var/folders/3t/xkwqrkld4xxgklk2s4n41jb80000gn/T/tmpeJR2pL/pycparser  pycparser 2.14ny.whl into distribution                                    
  cffi 1.7.0
pex: Building pex: 49.4ms                                                                                                                                                       
pex:   Resolving interpreter: 1.6ms
pex:     Setting up interpreter /Users/kwilson/Python/CPython-2.7.11/bin/python2.7: 1.6ms
pex:   Resolving distributions: 7.5ms
pex:     Fetching file:///private/tmp/unicode_wheels/test/cffi-1.7.0-cp27-cp27mu-macosx_10_10_x86_64.whl: 1.0ms
pex:       Fetching file:///private/tmp/unicode_wheels/test/cffi-1.7.0-cp27-cp27mu-macosx_10_10_x86_64.whl: 0.5ms
pex:     Translating /private/var/folders/3t/xkwqrkld4xxgklk2s4n41jb80000gn/T/tmpZVta6g/cffi-1.7.0-cp27-cp27mu-macosx_10_10_x86_64.whl into distribution: 1.1ms
pex:     Fetching file:///private/tmp/unicode_wheels/test/pycparser-2.14-py2-none-any.whl: 1.1ms
pex:       Fetching file:///private/tmp/unicode_wheels/test/pycparser-2.14-py2-none-any.whl: 0.7ms
pex:     Translating /private/var/folders/3t/xkwqrkld4xxgklk2s4n41jb80000gn/T/tmpeJR2pL/pycparser-2.14-py2-none-any.whl into distribution: 1.3ms
Saving PEX file to /tmp/throwaway.pex
```

for organizations with control over the contents of their `find-links` repos, the suggested workaround would be to eliminate all incompatible wheels from the resolve path - but we should fix this generally in pex.

**Comment by stuhood** [comment]
+1

**Comment by kwlzn** [comment]
this should now be mostly fixed as of #480.
